lib/rules/no-restricted-globals.js

```diff
@@ -100,20 +102,43 @@ module.exports = {
 			return Object.hasOwn(restrictedGlobalMessages, name);
 		}
 
+		/**
+		 * Check if the given reference occurs within a TypeScript type context.
+		 * @param {Reference} reference The variable reference to check.
+		 * @returns {boolean} Whether the reference is in a type context.
+		 * @private
+		 */
+		function isInTypeContext(reference) {
+			const parent = reference.identifier.parent;
+
+			return (
+				parent.type === "TSTypeReference" ||
+				parent.type === "TSInterfaceHeritage" ||
+				parent.type === "TSClassImplements"
+			);
```

Comment:
We can simplify this by creating a set containing these strings and then just doing `typeNodes.has(parent.type)` instead of doing multiple comparisons.